server.go

```diff
@@ -30,6 +30,10 @@ type Server struct {
 	AutoStream bool
 	// Enables automatic replay for each new subscriber that connects
 	AutoReplay bool
+
+	// Specifies the function to run when client register or un-register
+	OnRegister   func(str *Stream)
```

Comment:
I think calling these functions `OnSubscribe` and `OnUnsubscribe` would make more sense?